weather.py

```diff
@@ -61,6 +83,19 @@ def load_data_from_csv(csv_file):
 
 
 def find_min(weather_data):
+    if weather_data == []:
+        return ()
+    else:
+        min_temp = weather_data[0]
+        min_temp_idx = 0
+        counter = 0
+        for temp in weather_data:
+            if float(temp) <= float(min_temp):
+                min_temp = float(temp)
+                min_temp_idx = counter
+            counter += 1
```

Comment:
Nice approach, including initializing `min_temp` to the first datapoint in `weather_data`.